src/os/bluestore/BlueFS.cc

```diff
@@ -2582,6 +2602,11 @@ int BlueFS::_flush_and_sync_log(std::unique_lock<ceph::mutex>& l,
     if (!to_release[i].empty()) {
       /* OK, now we have the guarantee alloc[i] won't be null. */
       int r = 0;
+      if (forced_direct_writes == true) {
```

Comment:
AFAIU this block is orthogonal to direct writes - this is primarily intended to drop cached data, i.e. one that has been read.
Shouldn't this be controlled by another setting then?